marathon/src/main/scala/io/buoyant/marathon/v2/Api.scala

```diff
@@ -108,15 +116,23 @@ object Api {
   }
 
   object healthyIpAddress {
-    def unapply(task: Task): Option[String] = {
-      if (isHealthy(task)) {
-        ipAddress.unapply(task)
+    def unapply(extendedTask: TaskWithHealthCheckInfo): Option[String] = {
+      if (isHealthy(extendedTask)) {
+        ipAddress.unapply(extendedTask.task)
       } else {
         None
       }
     }
   }
 
+  object healthCheckCount {
+    def unapply(healthChecks: Option[Seq[HealthCheck]]): Option[Int] =
```

Comment:
scala's `map` function on the `Option` can be pretty handy in these situations. You can achieve the same `Option[Int]` method result like this:
```scala
def unapply(healthChecks: Option[Seq[HealthCheck]]): Option[Int] = healthChecks.map(_.size)
```

Reply:
This is an unapply method that we actually want to always match, so it should always return Some, as it does here.

A potentially less confusing way to implement this would be as a plain method:
```
def healthCheckCount(healthChecks: Option[Seq[HealthCheck]]): Int = healthChecks match {
  case Some(checks) => checks.size
  case None => 0
}
```

and then invoking this method directly instead of in as part of a case destructuring.

But this section of the code relies heavily on unapply methods so take it or leave it.